python/dolfinx/io/utils.py

```diff
@@ -276,10 +276,13 @@ def read_mesh(
         )
         return Mesh(msh, domain)
 
-    def read_meshtags(self, mesh, name, attribute_name="", xpath="/Xdmf/Domain"):
-        mt = super().read_meshtags(mesh._cpp_object, name, attribute_name, xpath)
+    def read_meshtags_by_name(self, mesh, name, attribute_name, xpath="/Xdmf/Domain"):
```

Comment:
Would it be better as one function and using default arg `attribute_name=None`?

Reply:
I remember that you don't like having two default arguments in one function but I'm guessing that's only for C++, right?

Reply:
Not good for C++, fine with Python if it's simple (which this is) and doesn't introduce complicated conditional handling.

Reply:
Done